python-frontend/src/main/java/org/sonar/python/types/TypeShed.java

```diff
@@ -276,6 +279,23 @@ public void visitFunctionDef(FunctionDef functionDef) {
       super.visitFunctionDef(functionDef);
     }
 
+    @Override
+    public void visitAnnotatedAssignment(AnnotatedAssignment annotatedAssignment) {
+      if (annotatedAssignment.variable().is(Tree.Kind.NAME)) {
+        Name variable = (Name) annotatedAssignment.variable();
+        Optional.ofNullable(variable.symbol()).ifPresent(symbol -> setAnnotatedType(symbol, annotatedAssignment));
+      }
+      super.visitAnnotatedAssignment(annotatedAssignment);
+    }
+
+    private static void setAnnotatedType(Symbol symbol, AnnotatedAssignment annotatedAssignment) {
+      TypeAnnotation typeAnnotation = annotatedAssignment.annotation();
+      if (symbol.is(Symbol.Kind.OTHER)) {
+        SymbolImpl other = (SymbolImpl) symbol;
+        other.setInferredType(InferredTypes.declaredType(typeAnnotation));
```

Comment:
I don't like that we use `setInferredType` for something that it's not inferred.
I think `SymbolImpl` should have a `typeAnnotation` setter, and we should set the symbol of `annotatedAssignment` instead of calling `InferredTypes.declaredType(typeAnnotation)`